setup.py

```diff
@@ -54,21 +55,37 @@
 test_reqs = ["pytest", "pytest-cov"]
 
 
-def get_version():
+def get_version(nightly_build=False):
     VERSIONFILE = join("pymc", "__init__.py")
     lines = open(VERSIONFILE).readlines()
     version_regex = r"^__version__ = ['\"]([^'\"]*)['\"]"
     for line in lines:
         mo = re.search(version_regex, line, re.M)
         if mo:
-            return mo.group(1)
+            version = mo.group(1)
+
+            if nightly_build:
+                from datetime import datetime, timezone
+
+                suffix = datetime.now(timezone.utc).strftime(r".dev%Y%m%d")
+                version += suffix
+
+            return version
+
     raise RuntimeError(f"Unable to find version in {VERSIONFILE}.")
 
 
+if "BUILD_PYMC_NIGHTLY" in os.environ:
```

Comment:
This whole logic block should be able to be replaced by `os.getenv` whose value is passed straight to `get_version`

https://docs.python.org/3/library/os.html#os.getenv

Reply:
Could you explain what you mean by this? `os.getenv` returns the value of the environment variable key if it exists, or default if it doesn’t, rather than True/False, so I don't see how `os.getenv("BUILD_PYMC_NIGHTLY")` can be passed directly into `get_version()`

Reply:
My apologies youre right. this can be done just fine with the if block you have above. My suggestion then is to move this entire block of logic to `get_version` and also consolidate the distname in there as well.

The reason for this is right now to understand where a version and dist name is coming from one needs to trace in four (and a half) places
1. Github secrets setting
2. The yml file
3. the `get_version` function
4.  This conditional in the global scope

In OSS the code that builds the package, while extremely important, often is not well understood by most people and is not tested, so its good to make it as simple as possible. That way when someone needs to change it its very clear and precise whats happening. Hope that helps!

Reply:
I've used two separate functions. Before editing the setup.py, we had `version=get_version()`, so this maintains one function for name and version